### PR TITLE
AKI-692: Refactor Sync manager to use ExecutorService 

### PR DESCRIPTION
**Pull request by AlexandraRoatis**
## Type of change

- [ ] Bug fix.
- [ ] New feature.
- [x] Enhancement.
- [ ] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

### CODE REVIEW

File: modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java [hunk 1]
```diff
@@ -380,6 +394,25 @@ private void interruptAndWait(Thread t, long timeout) {
...
 
+    private void shutdownAndAwaitTermination(ExecutorService pool) {
+        pool.shutdown(); // Disable new tasks from being submitted
+        try {
+            // Wait a while for existing tasks to terminate
+            if (!pool.awaitTermination(60, TimeUnit.SECONDS)) {
+                pool.shutdownNow(); // Cancel currently executing tasks
+                // Wait a while for tasks to respond to being cancelled
+                if (!pool.awaitTermination(60, TimeUnit.SECONDS)) {
+                    log.error("Pool did not terminate");
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java:459]: Should we throw the exception in this if condition?

File: modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java [hunk 2]
```diff
@@ -393,11 +439,32 @@ public long getNetworkBestBlockNumber() {
     }
 
     public synchronized void shutdown() {
+        if (p2pLog.isDebugEnabled()) {
```
Comment (AionJayT) [inline modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java:425]: minor: the check can be removed

File: modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java (hunk 1 above)
Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java:459]: Would an exception provide any additional information? I can add a stack trace to the log if you think it's useful?

File: modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java (hunk 2 above)
Comment (AlexandraRoatis) [reply modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java:425]: The check here is actually an optimization to skip the other if statements when debug is disabled. Would you still want it removed?

File: modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java (hunk 1 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java:459]: To throw the exception is not necessary, it mainly try to interrupt the thread. If the kernel thread stuck, we can dump the process threads info.

File: modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java (hunk 2 above)
Comment (AionJayT) [reply modAionImpl/src/org/aion/zero/impl/sync/SyncMgr.java:425]: I see, because you call the method in the log argument. Just ignore my comment.

**Pull request merged**